### ISSUE TITLE
semantic: no error happens with same identifier for imports and local variables

### ISSUE THREAD

**Issue by sapphi-red** [oxc-project/oxc#15961]
An error should be output for the following codes, but semantic does not report any errors:
```js
import { Foo } from './dep.js'

let Foo
```
[playground](https://playground.oxc.rs/#eNptU01v2zAM/SuBLr1kXVdgQJHb0C7AgG7N1mG95ELbdKpUFjWKTlIY+e+jHEvpgJ4sP4of7/FpMLVZGNsFYpkNsyXR7DhrmbrZxeXHBsPlNl6s/do7lBRcezM3ZBaD4d6nj7NezEK4x7lpiTsQQTaLFlxURBh8THBBbCQHgs0d1g4YxJKPJbh/toIxQI0F6sBv3Pm3pi4wxnNKrClo+DRAfO0qcvmvbjfTtePcBOCYBhsMHgR91L7Ku9tG5QPO0f4XSs/+oZdoG1z2vk6j5VKpJ/IOV8Cam9Ex78/NNy9stWJdAhF1bLH1V2biCdUZklbjDHp+o9Vg+oi/oTqTEqiebCPPZnE9N+ibh/beemVpXKvjBu0mU/zmSrvZJNHPnuSs0zYeHt+B/6a/FVPQXgbiB4/YYKM1dVFWx9vcUtdBijmnqPKwhaxy2Wf+Gt/Da9KuYqhfUB51aZqeL2cUOjxNPvWnaou1PDEErZFFTY3GWb+ICln1OiHyf2l4CMi2Qy/gHvQIQryiaE8rSgoZlbSY7SSqAG9QvWkwXl99+qxdVOY7bLXwkvjWQYxLi64pW3vbRe1JY5tztLNS0O8o0ICoUtNq37f1YKIyCmoQZA9qzOzGmhrc4PiAfDJC8ew2NqQPcpLce5KxVkYcbvLdsUp+DMlR+aUocwr3uMNS9AUx/NBVZIONqWpacks1cMrYIVcUs+CJj0+LSqFyOircjPIluJyOCU8yZ07Hf67Oiws=)
```js
import { Foo } from './dep.js'

const Foo = 1;
```
[playground](https://playground.oxc.rs/#eNptU01v2zAM/SuBLr1kXVtgQJFhh6FdgAHdmq3DesmFlulUqSxqFJ2kCPLfRzmW0wI9WX4UP97j095YMzOujcQy2U/mRJPDpGFqJ2fnH2uM5+t0tgzLYCkk6cNfJpefl8FMDZnZ3nAX8se7IGYm3OHUNMQtiCCbWQM+KSIMIWV4RFwiD4L1LVoPDOK0/BjcPjnBFMHiCLUQVv70a6mNjOmUkixFDR8HSC9tRb782WY1XDtMTQROebC9wZ1gSNpX6bfrpHzAe9r+Ruk43HeSXI3zLtg8WimVeyJvcAGsuQXt8/5efw/CTivaMZBQxxZnvzETD6jOkLXqZ9DzK632pkv4B6oTKYHq0dXyZGZXU4Ohvm/uXFCWxjc6btRuMsSvL7SbyxL96khOOq3T7uEd+F/+WzBF7WUgfQiINdZaUxfldLzVDbUt5Jj3iioPN5JVLtvCX+NbeMnaVQz2GeVBl6bp5XJBocXj5EN/qtZo5ZEhao0iam7Uz/pVVMiq0wmR36ThLiK7FoOAv9cjCPGCkjuuKCtkVNLRbEdRBXiF6k2D6eri8pN2UZlvsdHCc+IbDynNHfp63NrrLmpP6tucoq2TEf2BAjWIKjWs9n1b701SRlENghxAjVncaKnGFfYPKGQjjJ5dp5r0XQ6Sh0DS1yqIx1W521cpjyE7qrwUZU7xDjc4Fn1GjD91FcVgfaqalvxcDZwzNsgVpSJ45hPyonJoPB0Urnv5MjyeDhnPMhdOh/+SKozW)
```js
import { Foo } from './dep.js'

class Foo {}
```
[playground](https://playground.oxc.rs/#eNptU01v2zAM/SuBLr1kXVdgQJHb0C7AgG7N1mG95ELbtKtUFjWKTlIY/u+jHMvpgJ4sP4of7/GpN6VZGdsGYln0izXRYljUTO3i4vJjheFyFy+2futLBzGO4X7YerM0ZFa94c6nj7NezEq4w6WpiVsQQTarGlxURBh8TPCM2EgOBKs71LIMYsnHOXh4toIxQIkz1IJv3Pm3pDYwxnNKLClo+DRAfG0LcvmvrJvp2rA0ATimwXqDR0Efta+Sb3dR+YBzdPiF0rF/6CTaCtedL9NouVTqibzHDbDmZnTM+3PzzQtbrVjOgYg6ttjyKzPxhOoMSatxBj2/0ao3XcTfUJxJCRRPtpJns7peGvTVQ31vvbI0rtZxg3aTKX5zpd1skuhnR3LWaRePj+/Af9PfhiloLwPxg0essNKauiir4zW31LaQYs4pqjzsTFa5HDJ/jR/gNWlXMJQvKI+6NE3PlzMKLZ4mn/pTscNSnhiC1siipkbjrF9EhSw6nRD5vzQ8BmTbohdwD3oEId5QtKcVJYWMSjqb7SSqADeo3jQYr68+fdYuKvMd1lp4TXybTL226Kp5a2+7qD1pbHOOtlZm9DsKVCCq1LTa923dm6iMghoE2YMaM7uxpAobHB+QT0aYPbuLFemrnCT3nmSslRGHTb47VsmPITkqvxRlTuEe9zgXfUEMP3QV2WBjqpqW3FoNnDL2yAXFLHji49OiUmg+DQpXo3wJnk9DwpPMmdPwD+wJjPQ=)

When I run these on Node, I get `SyntaxError: Identifier 'Foo' has already been declared`.

**Comment by Boshen** [comment]
I tried 

```diff
diff --git a/crates/oxc_semantic/src/binder.rs b/crates/oxc_semantic/src/binder.rs
index 6e993430f..dd9ce7310 100644
--- a/crates/oxc_semantic/src/binder.rs
+++ b/crates/oxc_semantic/src/binder.rs
@@ -304,7 +304,9 @@ fn declare_symbol_for_import_specifier(
         ) {
         SymbolFlags::TypeImport
     } else {
-        SymbolFlags::Import
+        // Note 2 The LexicallyDeclaredNames of a Module includes the names of all of its imported bindings.
+        // <https://tc39.es/ecma262/#sec-static-semantics-lexicallydeclarednames>
+        SymbolFlags::Import | SymbolFlags::BlockScopedVariable
     };
```

but the interaction is super weird in conformance tests:

* TypeScript - declaration merging logic
* Test262 - no changes
* Babel - no error in https://github.com/babel/babel/tree/main/packages/babel-parser/test/fixtures/typescript/scope/redeclaration-import-let but the playground flags it

**Comment by sapphi-red** [comment]
I guess we have to allow this in TS files because we don't know whether the imported one is a value or a type.

**Comment by Boshen** [comment]
```diff
--- a/crates/oxc_semantic/src/builder.rs
+++ b/crates/oxc_semantic/src/builder.rs
@@ -470,7 +470,11 @@ impl<'a> SemanticBuilder<'a> {

         if report_error {
             let flags = self.scoping.symbol_flags(symbol_id);
-            if flags.intersects(excludes) {
+            if flags.intersects(excludes)
+                || (self.source_type.is_javascript()
+                    && flags.contains(SymbolFlags::Import)
+                    && excludes.contains(SymbolFlags::BlockScopedVariable))
+            {
                 let symbol_span = self.scoping.symbol_span(symbol_id);
                 self.error(redeclaration(name, symbol_span, span));
             }
```

Does not trigger anything in Test262. I'm confused.
